src/server/enterprise/cmds/cmds.go

```diff
@@ -192,6 +201,50 @@ func GetStateCmd() *cobra.Command {
 	return cmdutil.CreateAlias(getState, "enterprise get-state")
 }
 
+func SyncContextsCmd() *cobra.Command {
+	syncContexts := &cobra.Command{
+		Short: "Pull all available Pachyderm Cluster contexts into your pachctl config",
+		Long:  "Pull all available Pachyderm Cluster contexts into your pachctl config",
+		Run: cmdutil.Run(func(args []string) error {
+			cfg, err := config.Read(false, false)
+			if err != nil {
+				return err
+			}
+
+			c, err := client.NewOnUserMachine("user")
+			if err != nil {
+				return errors.Wrapf(err, "could not connect")
+			}
+			defer c.Close()
+			resp, err := c.License.ListUserClusters(c.Ctx(), &license.ListUserClustersRequest{})
+			if err != nil {
+				return err
+			}
+
+			// update the pach_address of all existing contexts, and add the rest as well.
+			for _, cluster := range resp.Clusters {
+				if context, ok := cfg.V2.Contexts[cluster.Id]; ok {
```

Comment:
I don't know if we need to check for an existing context, we could just blindly overwrite the existing value for the key?

Reply:
that works! 

my only concern was erasing any useful info that was already set for a context. Is there anything a user could manually set that we would want to preserve?

Reply:
Oh yeah, I completely forgot about the user's auth token, we probably want to keep that 😓 

Thinking about it a little - if the cluster deployment id has changed, the auth token is probably no longer valid and we should set it to an empty string (so the user is prompted to log in). If the deployment id hasn't changed we should keep the auth token so the user doesn't have to re-log-in every time they sync.